### PR TITLE
[optimize]split the code about resource event into event_handler.go

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
it is a code structure optimizition to split the code about resource event into event_handler.go

change **WaitForCacheSync**  to **s.informerFactory.WaitForCacheSync** for all imformer's resource sync and not just synchronization of  the  resource binding 
https://github.com/karmada-io/karmada/blob/361978a3bf339db62b27757eacb45cf8194135ac/pkg/scheduler/scheduler.go#L190-L192

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
NONE
**Does this PR introduce a user-facing change?**:
NONE
```release-note

```

### CODE REVIEW

File: pkg/scheduler/event_handler.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+	default:
+		klog.Errorf("cannot convert to clusterv1alpha1.Cluster: %v", t)
+		return
+	}
+	klog.V(3).Infof("Delete event for cluster %s", cluster.Name)
+
+	if s.enableSchedulerEstimator {
+		s.schedulerEstimatorWorker.Add(cluster.Name)
+	}
+}
```
Comment (RainbowMango) [inline pkg/scheduler/event_handler.go:275]: end of the line.

Comment (huone1) [reply pkg/scheduler/event_handler.go]: done, fixed

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -187,127 +139,18 @@ func (s *Scheduler) Run(ctx context.Context) {
 	}
 
 	s.informerFactory.Start(stopCh)
-	if !cache.WaitForCacheSync(stopCh, s.bindingInformer.HasSynced) {
-		return
-	}
+	s.informerFactory.WaitForCacheSync(stopCh)
```
Comment (XiShanYongYe-Chang) [inline pkg/scheduler/scheduler.go:142]: If there is no cacheSyncs to wait, does this line still need?

File: pkg/scheduler/event_handler.go [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/features"
+	informerfactory "github.com/karmada-io/karmada/pkg/generated/informers/externalversions"
+	"github.com/karmada-io/karmada/pkg/scheduler/metrics"
+	"github.com/karmada-io/karmada/pkg/util/gclient"
+)
+
+// addAllEventHandlers is a helper function used in tests and in Scheduler
+// to add event handlers for various informers.
+func addAllEventHandlers(
```
Comment (XiShanYongYe-Chang) [inline pkg/scheduler/event_handler.go:24]: The method can be directly define as:
```go
func (s *Scheduler) addAllEventHandlers
```

Comment (huone1) [reply pkg/scheduler/event_handler.go]: modified

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (huone1) [reply pkg/scheduler/scheduler.go:142]: if there is no cr object to wait， WaitForCacheSync will directly return

Comment (XiShanYongYe-Chang) [reply pkg/scheduler/scheduler.go:142]: ok, I got it.

**Review by RainbowMango — COMMENTED** [review]
LGTM
/cc @Garrybest @dddddai

**Comment by Garrybest** [comment]
Nice work!
/lgtm

**Comment by dddddai** [comment]
lgtm, this can make `scheduler.go` focus on scheduling

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**